langchain/agents/tools.py

```diff
@@ -74,19 +65,15 @@ async def _arun(self, tool_name: str) -> str:
 def tool(
     *args: Union[str, Callable],
     return_direct: bool = False,
-    args_schema: Optional[Type[BaseModel]] = None,
-    infer_schema: bool = True,
+    args_schema: Optional[Type[StringSchema]] = None,
```

Comment:
You could argue that we shouldn't add this. I like having the string args schema as required on the `BaseTool` since  making it optional means people have to always check for nones. The biggest potential source of confusion permitting user overrides of the string args schema may create would be if they add additional fields rather than overwriting the existing one I think